src/de.rs

```diff
@@ -398,6 +400,50 @@ impl<S: Read> Decoder<S> {
         Ok(i)
     }
 
+    pub(crate) async fn peek(&mut self) -> Result<Option<u8>, Error> {
```

Comment:
```suggestion
    async fn peek(&mut self) -> Result<Option<u8>, Error> {
```
might not need this level visibility

Reply:
Yes there's no decoding logic outside this module so I think this should be private